pyproject.toml

```diff
@@ -0,0 +1,37 @@
+[build-system]
+requires = [ "setuptools>=62", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "confluent-kafka"
+version = "2.3.0"
+description = "\"Confluent's Python client for Apache Kafka\""
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Topic :: Software Development :: Libraries :: Python Modules"]
+dependencies = [ "futures;python_version<\"3.2\"", "enum34;python_version<\"3.4\"",]
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.7"
+
+[[project.authors]]
+name = "\"Confluent Inc\""
+email = "support@confluent.io"
+
+[project.urls]
+Homepage = "https://github.com/confluentinc/confluent-kafka-python"
+
+[project.optional-dependencies]
```

Comment:
To avoid duplicating dependencies we can move all requirements.txt to the project root folder and use them as dynamic dependencies, like:


```
[project]
dynamic = ["dependencies", "optional-dependencies"]

[tool.setuptools.dynamic]
dependencies = {file = ["requirements.txt"]}
optional-dependencies.schemaregistry = { file = ["requirements-schemaregistry.txt"] }
optional-dependencies.avro = { file = ["requirements-avro.txt", "requirements-schemaregistry.txt"] }
```

Reply:
You cannot reference 
`-r requirements-schemaregistry.txt` in `requirements-avro.txt` so you can a `requirements-dev-install.txt` that lists the dependencies for the CI
```
-r requirements-schemaregistry.txt
-r requirements-avro.txt
-r requirements-json.txt
-r requirements-protobuf.txt
-r requirements-dev.txt
```

Reply:
We should also change tox.ini to use
`pip install .[dev]`
```
deps =
    -rrequirements-dev-install.txt
``` 
and add `py311`. `py312` lacks of the package `pkg_resources` needed by Trivup

Reply:
Also replace this in `DEVELOPER.md#Generate Documentation`

```
Install doc dependencies:

    $ pip install .[doc]
```

Reply:
You can use `requirements-dev-install.txt` when scripts are requiring different requirements files

Reply:
Once we install with `.[doc]` we can remove this import in `docs/conf.py`
`sys.path[:0] = [os.path.abspath(x) for x in glob('../build/lib.*')]`